src/MagicWormhole/Internal/Peer.hs

```diff
@@ -146,3 +147,10 @@ sendMessage conn body = do
 -- Obtain an 'EncryptedConnection' with 'withEncryptedConnection'.
 receiveMessage :: EncryptedConnection -> STM ClientProtocol.PlainText
 receiveMessage conn = snd <$> Sequential.next (inbound conn)
+
+
+-- | Derive a new key for the given purpose
+--
+-- Construct a new key from the existing session key for the given purpose
```

Comment:
maybe "... the existing encrypted connection's session key ..." here?

Reply:
Thanks, fixed in https://github.com/LeastAuthority/haskell-magic-wormhole/pull/24/commits/44fff28b530d2dc89851954f6d844d0c9d1d020d.